### ISSUE TITLE
[consistent-type-imports] False positive with decorator metadata

### ISSUE THREAD

**Issue by TomMettam** [typescript-eslint/typescript-eslint#4608]
- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

**Repro**

```
(no rule configuration, using plugin:@typescript-eslint/all)
```

```TS
import type { Ref } from '@typegoose/typegoose';
import { prop } from '@typegoose/typegoose';

export class Test
{
    @prop({ ref: () => Test })
    public test?: Ref<Test>;
}
```

**Expected Result**

`Ref` is only used as a type, and is not used by the decorator, no issue should be raised here.

**Actual Result**

`ESLint: Type import "Ref" is used by decorator metadata.(@typescript-eslint/consistent-type-imports)`

However, typescript compiles the code fine with 'importsNotUsedAsValues'.

When removing `type`, typescript compilation fails:

```
../../src/backend/models/Test.ts:1:1 - error TS1371: This import is never used as a value and must use 'import type' 
because 'importsNotUsedAsValues' is set to 'error'.

1 import { Ref } from '@typegoose/typegoose';
  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `13.1.0` |
| `@typescript-eslint/parser`        | `5.12.1` |
| `TypeScript`                       | `4.5.5` |
| `ESLint`                           | `8.10.0` |
| `node`                             | `16.13.0` |

**Comment by JoshuaKGoldberg** [comment]
Thanks for the report! It's good to know that this is a case that happens outside of Vue (#3108). 

cc @bradzacher - tentatively I think we're stuck here and will just have to take a breaking change on `consistent-type-imports` to use the type checker?

**Comment by bradzacher** [comment]
The lint rule *mostly* just relies upon what every is reported by the scope analysis.
#2751 added support for it to the scope analyser.

However, right now we only derive that config from the tsconfig - meaning you *need* type information to turn on this code path.
https://github.com/typescript-eslint/typescript-eslint/blob/9d47a8b94577540ae445e5ec17373da25423eeb8/packages/parser/src/parser.ts#L167-L170

We could definitely add a `parserOption` for this so that you can override turn it on without type information.

**Comment by TomMettam** [comment]
I don't quite understand the above, but I am using `emitDecoratorMetadata` in my tsconfig

**Comment by bradzacher** [comment]
> I don't quite understand the above, but I am using emitDecoratorMetadata in my tsconfig

if you are using [type-awawre linting](https://typescript-eslint.io/docs/linting/type-linting) then it will work out of the box.

**Comment by TomMettam** [comment]
The above issue was logged with type-aware linting enabled

**Comment by bradzacher** [comment]
Have you got a more complete example of the types you've imported?
It's hard to provide much guidance with just one file.

**Comment by bradzacher** [comment]
I believe this is a duplicate of https://github.com/typescript-eslint/typescript-eslint/issues/4268#issuecomment-987530004 slash https://github.com/typescript-eslint/typescript-eslint/issues/3108#issuecomment-785342161

The TL;DR is that TS uses type information to resolve that `Ref` is an object type, and emits `__metadata("design:type", Object)`.
We can't resolve this without using type information - so we cannot do this.

In general the decorator support is not amazing. We've done the best we can with the information we've got - but it's not perfect by any means.
